Zend/zend.h

```diff
@@ -140,6 +140,21 @@ struct _zend_class_entry {
 	zend_function *__callstatic;
 	zend_function *__tostring;
 	zend_function *__debugInfo;
+
+	/* magic functions for operator overloading */
+	zend_function *__add;
+	zend_function *__sub;
+	zend_function *__mul;
+	zend_function *__pow;
+	zend_function *__div;
+	zend_function *__concat;
+	zend_function *__mod;
+	zend_function *__sl;
+	zend_function *__sr;
+	zend_function *__or;
+	zend_function *__and;
+	zend_function *__xor;
```

Comment:
My two bits for better developer experience:

I suggest to use full names of methods, this is just not readable. `__add`, `__subtract`, `__multiply` etc. is much clearer.

Also, `on*` method naming convention came to my mind (`__onAddition`, `__onMultiplication` etc.) You can think about the operator as an event emitter (`$a + $b`) and the magic method as a handler. Not sure if it is a good idea, but I guess it is more explicit about when the method is run actually.